flake-lang/flake-rust.nix

```diff
@@ -17,10 +17,15 @@ pkgs:
 , cargoNextestExtraArgs ? ""
 }:
 let
+
   rustWithTools = pkgs.rust-bin.stable.${rustVersion}.default.override {
     extensions = [ "rustfmt" "rust-analyzer" "clippy" "rust-src" ];
   };
-  craneLib = crane.lib.${pkgs.system}.overrideToolchain rustWithTools;
+  craneLib =
+    if crane == null then
+      crane'.lib.${pkgs.system}.overrideToolchain rustWithTools
+    else
+      throw ''rustFlake: You're setting the `crane` argument which is deprecated and will be removed in the next major revision'';
```

Comment:
I don't think it's `throw` you want to only issue a `warning` (not sure if that's builtins.trace or smtn else) or else this will break everybody.

I think it's ok to ignore the `crane` the user did set and use the 'correct' one.